docker-compose.yaml

```diff
@@ -0,0 +1,79 @@
+version: "3.8"
+
+services:
+  nginx:
+    image: nginx:latest
+    container_name: rasalit-nginx
+    volumes:
+      - ./nginx/rasalit.conf:/etc/nginx/conf.d/default.conf
+      - ./nginx/index.html:/var/www/rasalit/html/index.html
+    ports:
+      - "8000:8000"
+    depends_on:
+      - rasalit-diet-explorer
+      - rasalit-live-nlu
+      - rasalit-nlu-cluster
+      - rasalit-overview
+      - rasalit-spelling
+    restart: on-failure
+
+  rasalit-diet-explorer:
+    build:
+      context: .
+      target: rasalit-diet-explorer
+    volumes:
+      - $RASA_PROJECT_DIR:/home/appuser
+    container_name: rasalit-diet-explorer
+    environment:
+      - PORT=8500
+
+  rasalit-live-nlu:
+    build:
+      context: .
+      target: rasalit-live-nlu
+    volumes:
+      - $RASA_PROJECT_DIR:/home/appuser
+    container_name: rasalit-live-nlu
+    depends_on:
+      - duckling
+    environment:
+      - PORT=8501
+      - "RASA_DUCKLING_HTTP_URL=http://duckling:8000"
+
+  rasalit-nlu-cluster:
+    build:
+      context: .
+      target: rasalit-nlu-cluster
+    volumes:
+      - $RASA_PROJECT_DIR:/home/appuser
+    container_name: rasalit-nlu-cluster
+    environment:
+      - PORT=8502
+
+  rasalit-overview:
+    build:
+      context: .
+      target: rasalit-overview
+    volumes:
+      - $RASA_PROJECT_DIR:/home/appuser
+    container_name: rasalit-overview
+    environment:
+      - PORT=8503
+      - FOLDER=${OVERVIEW_FOLDER:-gridresults}
+
+  rasalit-spelling:
+    build:
+      context: .
+      target: rasalit-spelling
+    volumes:
+      - $RASA_PROJECT_DIR:/home/appuser
+    container_name: rasalit-spelling
+    depends_on:
+      - duckling
+    environment:
+      - PORT=8504
+      - "RASA_DUCKLING_HTTP_URL=http://duckling:8000"
+
+  duckling:
+    image: rasa/duckling
+    container_name: rasalit-duckling
```

Comment:
I'm wondering if we need this around. If you're running duckling you should already have a config.yml file that is pointing to it. Is there a reason why we need it?

Reply:
I suppose not everyone has duckling in their NLU pipeline - I was using the financial-demo example which does have duckling, and the duckling server needs to be available to the "spelling" and "live-nlu" apps because they use `RasaNLUInterpreter` (if I understand correctly).

If you've got a duckling server running elsewhere on your machine, you can tell the apps where duckling is via the `RASA_DUCKLING_HTTP_URL` environment variable but the duckling server would need to be visible to the containers in the Rasalit Docker Compose, which means either using an external network or just using host networking in all of the Rasalit Docker Compose containers.

Reply:
I fear things might start getting tricky here.

If the duckling URL changes then the `config.yml` will require a retrain before you have a model that you can pass to rasalit. That means that suddenly you may need to train a model again (which is expensive) if you want to check it out via the docker-compose trick.

Reply:
I was under the impression that the `RASA_DUCKLING_HTTP_URL` always took precedence if present, meaning you don't need to retrain the model if the duckling URL changes - https://github.com/RasaHQ/rasa/issues/6240